store/auth.go

```diff
@@ -319,3 +258,48 @@ func requestDeviceSession(httpClient *http.Client, deviceSessionEndpoint string,
 
 	return responseData.Macaroon, nil
 }
+
+// retryPostRequestDecodeJSON calls retryPostRequest and decodes the response into either success or failure.
+func retryPostRequestDecodeJSON(httpClient *http.Client, endpoint string, headers map[string]string, data []byte, success interface{}, failure interface{}) (resp *http.Response, err error) {
+	return retryPostRequest(httpClient, endpoint, headers, data, func(resp *http.Response) error {
+		return decodeJSONBody(resp, success, failure)
+	})
+}
+
+// retryPostRequest calls doRequest and decodes the response in a retry loop.
+func retryPostRequest(httpClient *http.Client, endpoint string, headers map[string]string, data []byte, readResponseBody func(resp *http.Response) error) (*http.Response, error) {
+	return httputil.RetryRequest(endpoint, func() (*http.Response, error) {
+		req, err := http.NewRequest("POST", endpoint, bytes.NewBuffer(data))
+		if err != nil {
+			return nil, err
+		}
+		for k, v := range headers {
+			req.Header.Set(k, v)
+		}
+
+		return httpClient.Do(req)
+	}, readResponseBody, defaultRetryStrategy)
+}
+
+// a stringList is something that can be deserialized from a JSON
+// []string or a string, like the values of the "extra" documents in
+// error responses
+type stringList []string
+
+func (sish *stringList) UnmarshalJSON(bs []byte) error {
```

Comment:
what is `sish`? :-)

Reply:
this is just moved code, but I could tweak it. It's also not quite the right place for this either because I discovered it is used more than just in auth code